src/lib_ccaligner/grammar_tools.cpp

```diff
@@ -239,34 +239,38 @@ bool generate(std::string transcriptFileName, grammarName name) //Generate Gramm
         corpusDump.close();
     }
 
-    /*if (name == phone_lm || name == complete_grammar)
-    {
-        try
-        {
-            phoneticCorpusDump.open("tempFiles/corpus/phoneticCorpus.txt", std::ios::binary | std::ios::app);
-        }
-
-        catch (std::system_error& e)
-        {
-            FATAL(UnknownError, e.code().message().c_str());
-        }
-
-        int numberOfWords = transcript.length();
-        std::string printPhoneticCourpus = "SIL ";
-
-        for (int i = 0; i<numberOfWords; i++)
-        {
-            std::vector<Phoneme> phones = stringToPhoneme(stringToLower(sub->getWordByIndex(i)));
-
-            for (Phoneme ph : phones)
-                printPhoneticCourpus += ph + " ";
-        }
-
-        printPhoneticCourpus += "SIL\n";
-
-        phoneticCorpusDump << printPhoneticCourpus;
-        phoneticCorpusDump.close();
-    }*/ //Without phonetic transcription
+	if (name == phone_lm || name == complete_grammar)
+	{
+		try
+		{
+			phoneticCorpusDump.open("tempFiles/corpus/phoneticCorpus.txt", std::ios::binary | std::ios::app);
+		}
+
+		catch (std::system_error& e)
+		{
+			FATAL(UnknownError) << e.code().message();
+		}
+
+		std::istringstream iss(transcript);
+		std::vector<std::string> SplitTranscript((std::istream_iterator<std::string>(iss)),
+			std::istream_iterator<std::string>()); //split transcript into words
+
+		int numberOfWords = SplitTranscript.size();
```

Comment:
Careful there. `int` might not be big enough if `SplitTranscript` is huge. I don't know whether this will ever be the case, but just use `std::size_t` instead (of `auto` if you want to).